static/examples/batch/batch-slicing-threads.scd

```diff
@@ -0,0 +1,86 @@
+// Refer to https://learn.flucoma.org/overviews/batch-processing
+(
+s.waitForBoot({
+    var drum = Buffer.read(s,File.realpath(FluidDataSet.class.filenameSymbol).dirname.withTrailingSlash ++ "../AudioFiles/Nicol-LoopE-M.wav");
+    var slices = Buffer(s); // The slicer (FluidBufOnsetSlice) will write into this buffer the samples at which slices are detected
+    var data = FluidDataSet(s); // A FluidDataSet() for storing the analysis of each slice (mean centroid)
+
+    // Create a condition that hangs while processing is still happening and unhangs when its complete.
+    var processingCompleted = Condition();
+    var counter = 0; // Also keep track of how many slices we have processed.
+    s.sync;
+
+    // Slice the drum Buffer() based on consecutive changes in spectral frames
+    // The samples at which slices are detected will be written into the "slices" Buffer()
+    FluidBufOnsetSlice.processBlocking(s, drum, indices:slices, metric:9,threshold:0.5);
+
+    slices.loadToFloatArray(action:{ |slicesArr| // Bring the values in the slices buffer from the server to the language as a float array
+        slicesArr.postln;
+        slicesArr.doAdjacentPairs({ |start, end, sliceIndex|
+            /*
+            Take each of the adjacent pairs and pass them to this function as an array of 2 values
+
+            For example, [0,1,2,3,4] will execute this function 4 times, passing these 2 value arrays:
+            [0,1]
+            [1,2]
+            [2,3]
+            [3,4]
+
+            This will give us the left and right boundary of each segment
+            We can use this to inform the analysis (FluidBufSpectralShape) where to start analysing and how many frames to analyse
+            */
+
+            fork ({
```

Comment:
As with the other async example, this potentially starts a lot of threads on the server, which may be sub-optimal. 

As a side note, processing lots of very small slices isn't actually a great use case for threading as currently implemented, because we have to copy the whole source (rather than just the slice) into the new context every time. So, for a large source with lots of slices to process, it can end up being considerably slower than just ploughing through one by one. This is due to be fixed, but requires some internal redesign

Reply:
It's not a great case but I think it is good to show the ergonomics of it at least and to have parity with what exists as a Max patch too. I would opt to keep it there but add more information in the comments about the potential downsides due to copying etc.